src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -168,23 +165,30 @@ public function set($id, $service)
             throw new InvalidArgumentException('You cannot set service "service_container".');
         }
 
-        if (isset($this->aliases[$id])) {
-            unset($this->aliases[$id]);
-        }
-
-        $this->services[$id] = $service;
-
         if (null === $service) {
-            unset($this->services[$id]);
+            @trigger_error(sprintf('Unsetting a service ("%s") is deprecated since Symfony 3.2 and won\'t be supported anymore in Symfony 4.0.', $id), E_USER_DEPRECATED);
```

Comment:
the previous message was fine, and I'd keep the existing style for new messages too (`the "%s" service/alias` vs `a service (%s)`)